### PR TITLE
[Report Page] Add hover actions.

### PR DESCRIPTION
**Pull request by tfrcarvalho**
# Description

* Adds hover actions to the report page v2
* Fix a bug where it was showing an error when sample fails on experimental stage

# Tests

* Tested each of the actions

### CODE REVIEW

File: app/assets/src/components/views/SampleViewV2/ReportTable.jsx [hunk 1]
```diff
@@ -340,6 +354,137 @@ class ReportTable extends React.Component {
...
+    const phyloTreeEnabled =
+      phyloTreeAllowed &&
+      rowData.taxId > 0 &&
+      PhyloTreeChecks.passesCreateCondition(
+        getOr(0, "nt.count", rowData),
+        getOr(0, "nr.count", rowData)
+      );
+
+    const analyticsContext = {
+      projectId: projectId,
```
Comment (j-x-han) [inline app/assets/src/components/views/SampleViewV2/ReportTable.jsx]: Previous version of the report includes `projectName` as well, do we want to keep that?

File: app/assets/src/components/views/SampleViewV2/ReportTable.jsx [hunk 2]
```diff
@@ -462,20 +607,54 @@ class ReportTable extends React.Component {
...
+          rowClassName={cs.row}
+          rowRenderer={this.rowRenderer}
+          sortable={true}
+          sortedHeaderClassName={cs.sortedHeader}
+        />
+        {phyloTreeModalParams && (
+          <UserContext.Consumer>
+            {currentUser => (
+              <PhyloTreeCreationModal
+                admin={currentUser.admin}
```
Comment (j-x-han) [inline app/assets/src/components/views/SampleViewV2/ReportTable.jsx]: Some console errors when selecting phylo tree: 
image

File: app/services/pipeline_report_service.rb [hunk 3]
```diff
@@ -241,7 +247,11 @@ def fetch_taxon_counts(_pipeline_run_id, _background_id)
                                          tax_level: [TaxonCount::TAX_LEVEL_SPECIES, TaxonCount::TAX_LEVEL_GENUS]
                                        )
                                        .where.not(
-                                         tax_id: [TaxonLineage::BLACKLIST_GENUS_ID, TaxonLineage::HOMO_SAPIENS_TAX_ID]
+                                         tax_id: [
+                                           TaxonLineage::BLACKLIST_GENUS_ID,
+                                           TaxonLineage::HOMO_SAPIENS_TAX_ID,
```
Comment (j-x-han) [inline app/services/pipeline_report_service.rb]: It looks like `HUMAN_TAX_IDS` ([9605, 9606]) includes `HOMO_SAPIENS_TAX_ID` (9606), is there a reason why we have both?

File: app/assets/src/components/views/SampleViewV2/ReportTable.jsx (hunk 1 above)
Comment (tfrcarvalho) [reply app/assets/src/components/views/SampleViewV2/ReportTable.jsx]: No, it was conscious decision.

File: app/services/pipeline_report_service.rb (hunk 3 above)
Comment (tfrcarvalho) [reply app/services/pipeline_report_service.rb]: Actually no. Should have only HUMAN_TAX_IDS. Good catch.

File: app/assets/src/components/views/SampleViewV2/ReportTable.jsx [hunk 4]
```diff
@@ -1,34 +1,47 @@
...
   constructor(props) {
     super(props);
 
+    const { pipelineVersion } = props;
+
     this.state = {
       expandAllOpened: false,
       expandedGenusIds: new Set(),
       dbType: this.props.initialDbType,
+      // hover actions state
```
[the remark this answers is not in the document]
Comment (tfrcarvalho) [reply app/assets/src/components/views/SampleViewV2/ReportTable.jsx]: Removed.

File: app/assets/src/components/views/SampleViewV2/ReportTable.jsx [hunk 5]
```diff
@@ -462,20 +607,54 @@ class ReportTable extends React.Component {
...
+
+    if (xsrfCookies.length === 0) {
+      return null;
+    }
+    return decodeURIComponent(xsrfCookies[0].split("=")[1]);
+  };
+
   render = () => {
-    const { rowHeight } = this.props;
+    const { projectId, projectName, rowHeight } = this.props;
```
[the remark this answers is not in the document]
Comment (tfrcarvalho) [reply app/assets/src/components/views/SampleViewV2/ReportTable.jsx]: Thanks, fixed.

File: app/services/pipeline_report_service.rb (hunk 3 above)
Comment (tfrcarvalho) [reply app/services/pipeline_report_service.rb]: Will change `HOMO_SAPIENS_TAX_ID` to the correct definition. We should deprecate `HUMAN_TAX_IDS`.

File: app/assets/src/components/views/SampleViewV2/ReportTable.jsx (hunk 2 above)
Comment (tfrcarvalho) [reply app/assets/src/components/views/SampleViewV2/ReportTable.jsx]: Fixed.

**Pull request merged**